src/main/scala/strawman/collection/Factories.scala

```diff
@@ -22,35 +24,62 @@ trait FromSpecificIterable[-A, +C] extends Any {
   def fromSpecificIterable(it: Iterable[A]): C
 }
 
+/**
+  * Builder factory
+  * @tparam A Element type (e.g. `Int`)
+  * @tparam C Collection type (e.g. `List[Int]`)
+  */
+trait CanBuild[-A, C] {
+  /** Creates a builder */
+  def newBuilder(): Builder[A, C]
+}
+
 /** Base trait for companion objects of unconstrained collection types */
-trait IterableFactory[+CC[_]] extends FromIterable[CC] {
+trait IterableFactory[CC[_]] extends FromIterable[CC] {
   def empty[A]: CC[A]
 
   def apply[A](xs: A*): CC[A] = fromIterable(View.Elems(xs: _*))
 
   def fill[A](n: Int)(elem: => A): CC[A] = fromIterable(View.Fill(n)(elem))
+
+  def newBuilder[A](): Builder[A, CC[A]]
+
+  implicit def canBuild[A]: CanBuild[A, CC[A]] = () => newBuilder[A]()
```

Comment:
Some operations are inherently strict. This includes basic operations like grouping and sorting that make sense for all sequence types. I think it is generally desirable that they use the original collection type rather than always building a fixed type. Consequently we have to be able to build even a `LazyList` in a strict way. This requires copying the elements at some points but the same is true for other immutable collections and their builders. The result could still be a "proper" `LazyList`, albeit one whose elements have already been evaluated.